docs/DeveloperGuide.md

```diff
@@ -170,8 +170,13 @@ How the parsing works:
   placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse 
   the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `AddressBookParser` returns back as 
   a `Command` object.
-* All `XYZCommandParser` classes (e.g., `AddCommandParser`, `DeleteCommandParser`, ...) inherit from the `Parser` 
-  interface so that they can be treated similarly where possible e.g, during testing.
+* When parsing a `get` command, the `AddressBookParser` class creates a `GetCommandParser` to parse the prefix of the 
+  `get` command (e.g., `/hw`). If the `get` command only requires a prefix (e.g., `get /inp` & `get /outp`), the 
+  respective `GetXYZCommand` object is created. If the `get` command requires parameters (e.g., `get /hw North`), the
+  respective `GetXYZCommandParser` is created to parse the parameters and create the appropriate `GetXYZCommand` to be
+  returned.
+* All `XYZCommandParser` and `GetXYZCommandParser` classes (e.g., `AddCommandParser`, `DeleteCommandParser`, ...) 
+  inherit from the `Parser` interface so that they can be treated similarly where possible e.g, during testing.
```

Comment:
I think a better way to phrase it when our `get` commands require parameters is:
```
If the `get` command requires parameters (e.g., `get /hw North`), the prefix is parsed accordingly
within the `GetCommandParser` before the respective `GetXYZCommandParser` is created to
parse the parameters and create the appropriate `GetXYZCommand` to be returned.
```